### PR TITLE
Remove dependency of the legacy client code from pkg/cmd directory part 1

### LINKED ISSUE #6190: Investigate how to remove the generated legacy client-go code.
v1.12
- [x] Remove from `pkg/cmd` directory part 1 - #6469 
- [x] Remove from `pkg/cmd` directory part 2.

v1.13
- [x] Remove from `pkg/plugin` directory.
- [x] Remove from the rest directories.
- [x] Check with the community.
- [x] Record in the release note.

v1.15
- [ ] Remove the `pkg/generated` directory.
- [ ] Mention the change in the release note.

Backlog
- [ ] Update hack/test.sh to update the excluded packages.

**Describe the problem/challenge you have**

Since all Velero controllers already used the controller-runtime framework, Velero should consider whether to remove the legacy client-go code.
Investigate where those legacy codes are still used, and check whether it's possible to remove those codes.

**Describe the solution you'd like**

**Anything else you would like to add:**

**Environment:**

- Velero version (use `velero version`):
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

### PR DESCRIPTION
**Pull request by blackpiglet**
Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #6190

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: pkg/cmd/cli/backup/create.go [hunk 1]
```diff
@@ -107,7 +109,7 @@ type CreateOptions struct {
 	CSISnapshotTimeout              time.Duration
 	ItemOperationTimeout            time.Duration
 	ResPoliciesConfigmap            string
-	client                          veleroclient.Interface
+	client                          dynamic.Interface
```
Comment (ywk253100) [inline pkg/cmd/cli/backup/create.go:112]: Seems we have both the dynamic client and controller-runtime client in our code, could we just keep one of them? How about to use the controller-runtime client which is easy to use?

Comment (blackpiglet) [reply pkg/cmd/cli/backup/create.go:112]: The backup and restore create sub-command has a --wait option. Code uses an informer to watch the backup's change. The controller-runtime framework doesn't encourage users to use Informer with it. This is the main reason for choosing the dynamic client.

Comment (ywk253100) [reply pkg/cmd/cli/backup/create.go:112]: Just curious why `the controller-runtime framework doesn't encourage users to use Informer with it`?

Is it for the client provided by the controller manager which embeds informers? But for client we used in the CLI isn't the one provided by the controller manager, it has no informers embeded.

Comment (blackpiglet) [reply pkg/cmd/cli/backup/create.go:112]: https://github.com/kubernetes-sigs/controller-runtime/issues/1668
I found this issue, and the comments there made me feel it's not encouraged.
I agree that passing an informer not working doesn't mean the internal informer cannot be configured.
If using a customized source, and writing the data into a channel when the condition matches, it should also work.

The only downside is we need to write more boilerplate code for controller-runtime, such as manager and reconcile.

Comment (blackpiglet) [reply pkg/cmd/cli/backup/create.go]: Removed the dynamic client.

**Review by ywk253100 — DISMISSED** [review]

**Review by ywk253100 — APPROVED** [review]

**Review by Lyndon-Li — APPROVED** [review]

**Pull request merged**